beacon-chain/core/state/processing.go

```diff
@@ -66,3 +67,11 @@ func ProcessSpecialRecords(slotNumber uint64, validators []*pb.ValidatorRecord,
 	}
 	return validators, nil
 }
+
+// ProcessBlock describes the per block operations that happen on every slot.
+func ProcessBlock(state *types.BeaconState, block *types.Block) *types.BeaconState {
```

Comment:
Why do we have this and also a `processBlock` on the chainService?

Reply:
We should rename the processBlock on the chainService to something like "receiveBlock". ProcessBlock sounds more "spec-like" and a core part of the infrastructure and logic of state transitions, whereas the function in the chain service is more like receiving a block from some source and then passing it to the state transition and core functions.

Reply:
Alright changed the naming